### PR TITLE
Erase TWAI

### PR DESCRIPTION
**Pull request by bugadani**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] I have added necessary changes to user code to the [Migration Guide](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/MIGRATING-0.21.md).
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/documentation/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/documentation/CONTRIBUTING.md) and followed its instructions.

### CODE REVIEW

File: esp-hal/MIGRATING-0.21.md [hunk 1]
```diff
@@ -23,13 +23,34 @@ For example:
  }
 ```
 
+## Removed `async`-specific constructors
```
Comment (MabezDev) [inline esp-hal/MIGRATING-0.21.md:26]: What's the rationale behind this? Every other driver uses `new` & `new_async`? If we're going to make a change like this we need to be consistent.

Comment (bugadani) [reply esp-hal/MIGRATING-0.21.md:26]: I want to start working on https://github.com/esp-rs/esp-hal/issues/2321 and plan to touch up every other driver eventually

Comment (MabezDev) [reply esp-hal/MIGRATING-0.21.md:26]: Let's draft this until we flesh out exactly what we want to do in #2321

Comment (bugadani) [reply esp-hal/MIGRATING-0.21.md]: Undid those bits and I'll send in a follow-up later

File: esp-hal/src/twai/mod.rs [hunk 2]
```diff
@@ -980,46 +988,90 @@ where
...
+                twai: unsafe { self.twai.clone_unchecked() },
                 phantom: PhantomData,
             },
+            twai: unsafe { self.twai.clone_unchecked() },
             phantom: PhantomData,
         }
     }
+
+    /// Convert the configuration into an async configuration.
+    fn into_async(self) -> TwaiConfiguration<'d, Async, T> {
```
Comment (MabezDev) [inline esp-hal/src/twai/mod.rs:1026]: Remove this for now

**Review by MabezDev — CHANGES_REQUESTED** [review]
Missed a function, but after that LGTM

File: esp-hal/src/twai/mod.rs (hunk 2 above)
Comment (bugadani) [reply esp-hal/src/twai/mod.rs:1026]: https://github.com/esp-rs/esp-hal/pull/2359/files#diff-d8403089e88b2df8eb0be3db73af0ff946613a6b5be289d7dfb7162296551597R1144

**Review by MabezDev — APPROVED** [review]

**Review by bjoernQ — APPROVED** [review]
LGTM

**Pull request merged**